### PR TITLE
Filter emoji with CSS

### PR DESCRIPTION
**Pull request by lolmaus**
Great performance boost while typing!

### CODE REVIEW

File: addon/components/emoji-picker.js [hunk 1]
```diff
@@ -90,14 +98,11 @@ export default Component.extend(ClickOutsideMixin, {
...
 
-  filteredEmojiCount: computed(
-    `emojiByCategoryIdFiltered.{${DEPENDENT_KEYS_FOR_EMOJI_CATEGORIES}}.length`,
-    function () {
-      return EMOJI_CATEGORIES_ARRAY.reduce((count, category) => {
-        return count + this.get(`emojiByCategoryIdFiltered.${category}.length`);
-      }, 0);
-    }
-  ),
+  filteredEmojiCount: computed('emojiByCategoryIdFiltered', function () {
```
Comment (lolmaus) [inline addon/components/emoji-picker.js:101]: Removed the `DEPENDENT_KEYS_FOR_EMOJI_CATEGORIES` constant here.

The `emojiByCategoryIdFiltered` content is never changed except by creating a whole new `emojiByCategoryIdFiltered` object.

File: tests/integration/components/emoji-picker-test.js [hunk 2]
```diff
@@ -248,13 +248,13 @@ test('it should filter emoji', withChai(async function(expect) {
...
   m = "Initially should contain 1369 emoji";
-  expect(component.emoji().count).equal(1369);
+  expect(component.emoji().filterBy('isVisible')).length(1369);
 
   component.filterInput.fill("fo");
   await wait();
 
   m = "Should contain 93 emoji after filling 'fo' into search field";
-  expect(component.emoji().count).equal(93);
+  expect(component.emoji().filterBy('isVisible')).length(93);
```
Comment (lolmaus) [inline tests/integration/components/emoji-picker-test.js:257]: Page Object is awesome!

File: addon/components/emoji-picker.js (hunk 1 above)
Comment (vvainio) [reply addon/components/emoji-picker.js:101]: Makes sense 👍

File: addon/components/emoji-picker.js [hunk 3]
```diff
@@ -212,14 +210,13 @@ export default Component.extend(ClickOutsideMixin, {
...
-            emojo = emoji[0];
-            return true;
-          }
+      const emojiByCategoryIdFiltered = this.get('emojiByCategoryIdFiltered');
+
+      EMOJI_CATEGORIES_ARRAY
+        .map(categoryName => emojiByCategoryIdFiltered.get(categoryName))
+        .filter(categoryHash => get(categoryHash, 'visibleCount'))
+        .find(categoryHash => {
+          return emojo = get(categoryHash, 'emoji').find(emojo => get(emojo, 'isVisible'));
```
Comment (vvainio) [inline addon/components/emoji-picker.js:219]: Selecting by enter works now, good catch!

**Review by vvainio — APPROVED** [review]

**Pull request merged**